Frontend #155

APfrom3 opened the pull request:
All the changes for task number 155 as well as branch 135 and 164(Deepak) have been aggregated into this branch. This includes adding quickviews to database, adding API endpoints to controller and front end functions to fetch, display and remove quickviews.

Commit:
chnges include backend api calls and datbasse table creation for adding and getting the user quickviews for card 155'

Commit:
Merge branch 'develop' into #155Backend_Deepak

kartike2001 committed:
Filtering out the Multiselect working- kitname kitversion, created dependecy between filters

Commit:
Card number 164 changed the source of the /list api call to local database rather than vault api call

Commit:
Implemented front end functionality for back end functions , Addquickview and Getquickviews.

Commit:
Spelling error

Commit:
Implemented front end functionality for back end functions, AddQuick…
…View and GetQuickViews.

Commit:
WIP committing to save the changes ,changes needs to be tested waiting for the vault api error to resolve before testing

Commit:
updated the sql file to have UNIQUE constarint for name and username column in quickviews table

Commit:
Contains changes for new delete api call to delete quickk view for a user from db for the task 155

Commit:
changes include logic to compare the deployments between kit_details table and deployments_details_github table for the task #164

Commit:
Frontend quickviews are able to be added and be displayed from database rather than localstorage

Commit:
Merge fixes

Commit:
Mergeing delete quickview from DB changes to integrate into frontend

Commit:
Delete qv and Delete all qv's fully functional with token check

antitux commented on backend/controllers/user_controller.go:
```diff
@@ -29,10 +29,21 @@ type dataExtract struct {
-	client_secret = os.Getenv("CLIENT_SECRET")
+	//client_id     = os.Getenv("CLIENT_ID")
+	//client_secret = os.Getenv("CLIENT_SECRET")
+	client_secret="27c5646e9c70d276c4b487fa9f5d17366f31f2b1"
+	client_id="cfa78f27867826309fb6"
```
Secrets should never be hard coded in applications.

antitux replied:
These are something that should probably be read out of the `.env` file for local development. In production these would be set as environment variables at runtime. Also make sure the documentation is updated to add these parameters.

antitux commented on client/src/Settings.js:
```diff
@@ -1,6 +1,6 @@
 let Settings = {
     currentPort: "http://localhost:3000",
-    githubClientId: "d8ca7de576a6e29f75ca",
+    githubClientId: "cfa78f27867826309fb6",
```
This should be parameterized also.

antitux replied:
These are something that should probably be read out of the `.env` file for local development. In production these would be set as environment variables at runtime. Also make sure the documentation is updated to add these parameters.

deepak0797 replied:
updated the file to read from the env file in new commit so resolving this

antitux reviewed (changes requested):
We shouldn't be hard coding secrets inside source code. We need to figure out how to get these out of the source code and being read in from environment variables.

Commit:
updated the user_controller.go file to red the client id and secret from ENV variables and updated bin/docker to expose the env variable for client id and secret

Commit:
updated setting.js file to read the clinet id from env variable

deepak0797 committed:
Delete .env

the env file with tokens got pushed to th ebranch with other files so deleting this file

kartike2001 committed:
Fixed every behavior all functions are working, no bugs seen at the moment, every filter is working as desired. In dashboard implemented quicksort to sort kitVersion

Commit:
changes made to the dockerfile to remove the go get command and the commit has new modules added to go.mod

Commit:
updated code to read the github token from env file

Commit:
updated the setting.js file to read the client id from environment variable

Commit:
Updated the api function to remove extra space while adding values to db

Commit:
resaved these files to apply proper formatting for Golang

kartike2001 committed:
Fixed only single date selected behavior

kartike2001 committed:
Fixed a bug - Date Selected -> Kit Name Selected -> Nos of days selected

kartike2001 committed:
Fixed a bug - Date Selected -> Kit Name Selected -> Nos of days selected and Removed Comments

Commit:
Removed .replace of spaces in rendering of quickviews. Problem was foxed in the database

kartike2001 committed:
Removed Extra Variable

kartike2001 committed:
Removed all console logs

kartike2001 committed:
Removed all console logs

Commit:
Merging of branch task 138 and branch task 155

Commit:
updated the api backend logic to pull the deployment details from valut and update the db

antitux commented:
There are some merge conflicts that need to be resolved before this can be merged.

devikaranade committed:
Merge branch 'develop' into Frontend-#155
